### ISSUE TITLE
No longer supported in Safari 12

### ISSUE THREAD

**Issue by Juhannuspukki** [rfletcher/safari-json-formatter#36]
"Safari no longer supports unsafe extensions.

“JSON Formatter” is no longer supported. You can find newer extensions reviewed by Apple in the App Store or Safari Extensions Gallery."

**Comment by ruchernchong** [comment]
@rfletcher I would like to continue this project as I have an Apple Developer account to play around with it. I wish to get your permission before continuing.

**Comment by dankhazipeter** [comment]
Except the autorun, it works for me: https://georgegarside.com/blog/macos/install-any-safari-extension-macos-mojave/

**Comment by jespernissen** [comment]
I downloaded the source and build a new package using my company's signature.
It can be downloaded here:  http://cloud.wnb.dk/b337fd14f706/Safari%20JSON%20Formatter.safariextz

I hope the author doesn't mind, it unfortunately seems like he's inactive in this project.

**Comment by ruchernchong** [comment]
@roniekoma the only downside to this is whenever Safari is restarted, you have to redo the same steps again.

**Comment by rogierlommers** [comment]
Yes please!! Really mis it! You guys have any alternatives maybe?

**Comment by mcdemarco** [comment]
It doesn't work at all for me after following the instructions; the extension seems to "run" but nothing gets prettified and the toggle button doesn't appear.  I'm using the 1.1 zip from the release directory, MacOS 10.12.6, and Safari 12.0.  Fortunately I still have a copy of Safari Technology Preview (now a postview) that still runs the extension, but constantly complains about being out of date.

**Comment by nirodg** [comment]
It work for me with macOS Sierra (v10.12.6) and Safari Version 12.0 (12606.2.11)

**Comment by OlafGoral** [comment]
I couldn’t get this to work either, but I found one on the appstore, https://itunes.apple.com/us/app/asini-json-viewer/id1440033430?ls=1&mt=12 it’s working ok for me. I know it’s not the same as this one, but I found it and works fine for me.

**Comment by ayman** [comment]
Indeed many of us (myself included) would like to officially put this in the Apple store such that it's free and community supported. @rfletcher, can you put a friendly open license on this?  I'd also be happy to be added to the maintainers of this repo to help triage bugs and what not.

**Comment by ibrunotome** [comment]
The last commit was in 2012, I don't think that he will accept any changes :(

**Comment by ayman** [comment]
Probably just worth rewriting then!

**Comment by nirodg** [comment]
> Probably just worth rewriting then!

Agreed 

> Indeed many of us (myself included) would like to officially put this in the Apple store such that it's free and community supported. @rfletcher, can you put a friendly open license on this? I'd also be happy to be added to the maintainers of this repo to help triage bugs and what not.

Not such a bad idea, such a pity there is no license specified, could be forked and make a release version for the Apple Store.

**Comment by wuf810** [comment]
Hey @rfletcher you there? 

Would be great if you could respond to this:

https://github.com/rfletcher/safari-json-formatter/issues/36#issuecomment-437057297

Thanks, M

**Comment by nirodg** [comment]
> Hey @rfletcher you there?
> 
> Would be great if you could respond to this:
> 
> [#36 (comment)](https://github.com/rfletcher/safari-json-formatter/issues/36#issuecomment-437057297)
> 
> Thanks, M

I've tried to reach him on Twitter but no response so far

**Comment by ayman** [comment]
Ok. So there are a few new extensions in the Safari Extension store that do pretty formatting (new extensions have to be written in Xcode...these legacy ones are deprecated as of this month); a few of them are free too. But, I went ahead and rewrote one from scratch that is free and open to community contributions.  I've just thrown the v1 to the store (under review now). Check it out...contributions welcome: https://github.com/ayman/PrettyJSON

**Comment by quiram** [comment]
Thanks @ayman! Please let us know when it has passed review so we can install it :)

**Comment by ArtFeel** [comment]
Hitting **⌥⌘U** will open page sources. That gives you pretty good JSON formatter.  

<img width="1006" alt="image" src="https://user-images.githubusercontent.com/502063/54438660-c2e7e580-473f-11e9-9e7a-0e6e3127143e.png">

**Comment by ArtFeel** [comment]
There is also [JSON-Lite](https://safari-extensions.apple.com/details/?id=com.litejs.json-lite-YVKYWJZ9CZ) extension, it is working without a standalone mac app.

**Comment by 19h** [comment]
Just because there's so much non-free spam, this one is pretty good and free: https://apps.apple.com/de/app/json-peep-for-safari/id1458969831?l=en&mt=12.

**Comment by ruchernchong** [comment]
@ArtFeel I cannot find that in the App Store on macOS Catalina.

**Comment by ArtFeel** [comment]
@ruchern `JSON-Lite` extension? This is not standalone app, so it is unavailable in the App Store, but you can install it from URL above.  

If you mean `Developer Console`, you should enable Developer mode in Safari settings.
